### PR TITLE
feat/studio/logs: Custom sql querying

### PR DESCRIPTION
**Pull request by Ziinc**
This PR introduces custom SQL querying.

Most of the api legwork has already been implemented, so what remains is adjusting the UI for dynamic result sets.

Changes required:
- [x] adjust query param building to include `sql` param
- [x] adjusting LogTable for dynamic columns 
- [x] adding in restriction alerts (7 day for edge, 2 hr for postgres)
- [x] hiding chronological features
  - [x] hide load older btn
  - [x] hide selection preview
  - [x] first column should not be highlighted green

### CODE REVIEW

**Comment by Ziinc** [comment]
Trying to get infra running locally for demo vid, but code is ready to review in meantime.

File: studio/components/interfaces/Settings/Logs/LogTable.tsx [hunk 1]
```diff
@@ -9,33 +9,43 @@ import { isNil } from 'lodash'
...
   const [focusedLog, setFocusedLog] = useState<LogData | null>(null)
-  const columns = ['timestamp', 'event_message'].map((v) => ({
+  const columnNames = Object.keys(data[0] || {})
+  // whether it the data structure is LogData format.
+  const isChrono =
+    columnNames.includes('timestamp') &&
+    columnNames.includes('event_message') &&
+    columnNames.length === 4
+
+  const columns = (isChrono ? ['timestamp', 'event_message'] : columnNames).map((v) => ({
```
Comment (kiwicopple) [inline studio/components/interfaces/Settings/Logs/LogTable.tsx:30]: I had to jump back to figure out what `isChrono` is

Perhaps something more descriptive like `hasLogDataFormat` or similar?

File: studio/components/interfaces/Settings/Logs/LogTable.tsx [hunk 2]
```diff
@@ -9,33 +9,43 @@ import { isNil } from 'lodash'
...
  * Logs table view with focus side panel
+ *
+ * When in custom data display mode, the side panel will not open when focusing on logs.
  */
-const LogTable = ({ isCustomQuery, data }: Props) => {
+const LogTable = ({ isCustomQuery, data = [] }: Props) => {
   const [focusedLog, setFocusedLog] = useState<LogData | null>(null)
-  const columns = ['timestamp', 'event_message'].map((v) => ({
+  const columnNames = Object.keys(data[0] || {})
+  // whether it the data structure is LogData format.
```
Comment (kiwicopple) [inline studio/components/interfaces/Settings/Logs/LogTable.tsx:24]:
```suggestion
  // whether the data structure is LogData format.
```

File: studio/components/interfaces/Settings/Logs/LogTable.tsx [hunk 3]
```diff
@@ -44,36 +54,42 @@ const LogTable = ({ isCustomQuery, data }: Props) => {
...
+    const logData = data as LogData[]
+    return logData.reduce((acc: LogMap, d: LogData) => {
       acc[d.id] = d
       return acc
-    }, {})
+    }, {}) as LogMap
   }, [JSON.stringify(data)])
 
-  const strLogMap = JSON.stringify(logMap)
+  const strData = JSON.stringify(data)
```
Comment (kiwicopple) [inline studio/components/interfaces/Settings/Logs/LogTable.tsx:66]: full variable names are preferable. 

```suggestion
  const stringData = JSON.stringify(data)
```

File: studio/tests/pages/projects/LogTable.test.js [hunk 4]
```diff
@@ -17,17 +23,35 @@ test('can display log data', async () => {
...
-    timestamp: 1621323232312,
-    event_message: "some event happened",
-  }]} />)
+test('dedupes log lines with exact id', async () => {
+  // chronological mode requires 4 columns
+  render(
+    <LogTable
+      data={[
+        {
+          id: 'seome-uuid',
```
Comment (kiwicopple) [inline studio/tests/pages/projects/LogTable.test.js:32]:
```suggestion
          id: 'some-uuid',
```

File: studio/tests/pages/projects/LogTable.test.js [hunk 5]
```diff
@@ -17,17 +23,35 @@ test('can display log data', async () => {
...
+          timestamp: 1621323232312,
+          event_message: 'some event happened',
+          metadata: {}
+        },
+        {
+          id: 'seome-uuid',
```
Comment (kiwicopple) [inline studio/tests/pages/projects/LogTable.test.js:38]:
```suggestion
          id: 'some-uuid',
```

File: studio/components/interfaces/Settings/Logs/LogTable.tsx (hunk 1 above)
Comment (Ziinc) [reply studio/components/interfaces/Settings/Logs/LogTable.tsx:30]: changed, yes much more intuitive 😅

File: studio/components/interfaces/Settings/Logs/LogTable.tsx (hunk 3 above)
Comment (Ziinc) [reply studio/components/interfaces/Settings/Logs/LogTable.tsx:66]: done

**Comment by Ziinc** [comment]
Custom sql demo, with a simple count. 

https://user-images.githubusercontent.com/22714384/148185165-74fcafa8-e8b9-4ad2-bac5-b7b7584e3203.mov

Ready to merge, feature is hidden with the `logsCustomSql` feature flag.

**Comment by kiwicopple** [comment]
this is awesome @Ziinc - love the auto-detection of custom queries

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**